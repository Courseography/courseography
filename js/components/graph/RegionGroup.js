import PropTypes from "prop-types"
import React from "react"

// This now uses the new syntax for a stateless React component
// (component with only a render method).
export default function RegionGroup({ regionsJSON, labelsJSON }) {
  return (
    <g id="regions">
      {regionsJSON.map(function (entry, value) {
        var pathAttrs = { d: "M" }
        entry.points.forEach(function (x) {
          pathAttrs["d"] += x[0] + "," + x[1] + " "
        })

        var pathStyle = { fill: entry.fill }
        return <path {...pathAttrs} key={value} className="region" style={pathStyle} />
      })}
      {labelsJSON.map(function (entry, value) {
        var textAttrs = {
          x: entry.pos[0],
          y: entry.pos[1],
        }

        var textStyle = { fill: entry.fill }

        return (
          <text
            {...textAttrs}
            key={value}
            style={textStyle}
            className="region-label"
            textAnchor={entry["text-anchor"]}
          >
            {entry["text"]}
          </text>
        )
      })}
    </g>
  )
}

RegionGroup.propTypes = {
  labelsJSON: PropTypes.array,
  regionsJSON: PropTypes.array,
}
