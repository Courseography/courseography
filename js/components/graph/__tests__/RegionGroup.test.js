import React from "react"
import { shallow } from "enzyme"

import RegionGroup from "../RegionGroup"

describe("RegionGroup", () => {
  it("should match shallow snapshot", () => {
    const props = {
      regionsJSON: [
        {
          graph: 1,
          points: [
            [17.386348, 281.07376883],
            [17.386348, 14.41568908],
            [316.385978, 14.41576883],
            [316.386348, 281.07376883],
          ],
          isRegion: true,
          stroke: "",
          fill: "#6276b9",
          id_: "p83",
          source: "",
          target: "",
        },
      ],
      labelsJSON: [
        {
          graph: 1,
          rId: "tspan4346-9",
          text: "Systems",
          pos: [1088.1677413999998, 201.36453113000002],
          fill: "#000000",
          align: "begin",
        },
      ],
    }

    const wrapper = shallow(<RegionGroup {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
