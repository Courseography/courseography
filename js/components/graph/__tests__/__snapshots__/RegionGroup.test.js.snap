// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionGroup should match shallow snapshot 1`] = `
<g
  id="regions"
>
  <path
    className="region"
    d="M17.386348,281.07376883 17.386348,14.41568908 316.385978,14.41576883 316.386348,281.07376883 "
    key="0"
    style={
      Object {
        "fill": "#6276b9",
      }
    }
  />
  <text
    className="region-label"
    key="0"
    style={
      Object {
        "fill": "#000000",
      }
    }
    x={1088.1677413999998}
    y={201.36453113000002}
  >
    Systems
  </text>
</g>
`;
